src/features/auth/hooks/useRedirectAfterLogin.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect } from "react";
+
+import qs from "query-string";
+import { useHistory, useLocation } from "react-router-dom";
+
+import { useAppSelector } from "@app/redux/store";
+import { RedirectDef } from "@app/types/route.types";
+
+function useRedirectAfterLogin() {
+  const history = useHistory();
+  const location = useLocation<RedirectDef>();
+  const { redirect } = qs.parse(location.search);
```

Comment:
Query search params like `/settings/users?page=2&pageSize=6` is not handled here. It's only deal with `/settings/users`. Not the `?page=2&pageSize=6` etc. Not sure `useLocation` have support for this or not.

Reply:
Agreed, lets review this later, at least merge this in, so we have something in place that everyone can use.

Reply:
This is fixed